packages/tools/viewer/src/viewerFactory.ts

```diff
@@ -62,7 +62,22 @@ export async function CreateViewerForCanvas(
     canvas: HTMLCanvasElement,
     options?: CanvasViewerOptions & { viewerClass?: new (...args: ConstructorParameters<typeof Viewer>) => Viewer }
 ): Promise<Viewer> {
-    options = { ...defaultCanvasViewerOptions, ...options };
+    if (options) {
+        options = new Proxy(options, {

```

Comment:
`Proxy` is hurting performance, in case this options are being constantly polled.

Reply:
This should only be read on a call to reset, so there should not be any significant perf impact. My preference would be to leave it as is to keep maintenance easy and minimize the amount of actual code.